### PR TITLE
Created uber-jar test for multi-module projects

### PR DESCRIPTION
**Pull request by Garima829**

### CODE REVIEW

File: integration-tests/gradle/src/test/resources/uber-jar-for-multi-module-project/application/src/main/resources/META-INF/resources/index.html [hunk 1]
```diff
@@ -0,0 +1,155 @@
+<!DOCTYPE html>
```
Comment (geoand) [inline integration-tests/gradle/src/test/resources/uber-jar-for-multi-module-project/application/src/main/resources/META-INF/resources/index.html:1]: You can remove this file as it's not needed

Comment (Garima829) [reply integration-tests/gradle/src/test/resources/uber-jar-for-multi-module-project/application/src/main/resources/META-INF/resources/index.html:1]: Okk

File: integration-tests/gradle/src/test/java/io/quarkus/gradle/MultiModuleUberJarTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            }, output, ConditionTimeoutException.class);
+
+            String logs = FileUtils.readFileToString(output, "UTF-8");
+
+            assertThatOutputWorksCorrectly(logs);
+
+            // test that the http response is correct 
+            assertThat(DevModeTestUtils.getHttpResponse("/hello", () -> {
+                return jarRun == null ? null : jarRun.isDone() ? "jar run mode has terminated" : null;
+            }).equals("hello"));
```
Comment (geoand) [inline integration-tests/gradle/src/test/java/io/quarkus/gradle/MultiModuleUberJarTest.java:51]: If this is the actual response, then something is wrong...

```
@GET
    @Produces(MediaType.TEXT_PLAIN)
    public String hello() {
        return "hello " + common.getName();
    }
```

so the response should be `hello common`

Comment (Garima829) [reply integration-tests/gradle/src/test/java/io/quarkus/gradle/MultiModuleUberJarTest.java:51]: Oh the response was ```hello common``` only on localhost/8080 , I'll check this again

Comment (geoand) [reply integration-tests/gradle/src/test/java/io/quarkus/gradle/MultiModuleUberJarTest.java]: Did you now run the test and make sure it passes?

Comment (Garima829) [reply integration-tests/gradle/src/test/java/io/quarkus/gradle/MultiModuleUberJarTest.java]: Yes I did, and weird thing is I checked this earlier too, and it was passing then as well, even though the response on http://0.0.0.0:8080/hello was ```hello common``` both the times

Comment (geoand) [reply integration-tests/gradle/src/test/java/io/quarkus/gradle/MultiModuleUberJarTest.java]: OK, I'll run it later on and make sure everything is OK

Comment (Garima829) [reply integration-tests/gradle/src/test/java/io/quarkus/gradle/MultiModuleUberJarTest.java]: Should I mark it ready for review ?

Comment (geoand) [reply integration-tests/gradle/src/test/java/io/quarkus/gradle/MultiModuleUberJarTest.java]: Sure

**Comment by Garima829** [comment]
Had forgotten to format the code, so just did that

File: integration-tests/gradle/src/test/java/io/quarkus/gradle/MultiModuleUberJarTest.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.io.FileUtils;
+import org.awaitility.core.ConditionTimeoutException;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.devmode.util.DevModeTestUtils;
+
+public class MultiModuleUberJarTest extends QuarkusGradleWrapperTestBase {
+    private static Future<?> jarRun;
```
Comment (geoand) [inline integration-tests/gradle/src/test/java/io/quarkus/gradle/MultiModuleUberJarTest.java:20]: This seems like it's not used

Comment (geoand) [reply integration-tests/gradle/src/test/java/io/quarkus/gradle/MultiModuleUberJarTest.java:20]: Nevermind, my mistake

File: integration-tests/gradle/src/test/java/io/quarkus/gradle/MultiModuleUberJarTest.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+                return jarRun == null ? null : jarRun.isDone() ? "jar run mode has terminated" : null;
+            }).equals("hello common"));
+        } finally {
+            process.destroy();
+        }
+    }
+
+    private void assertThatOutputWorksCorrectly(String logs) {
+        assertThat(logs.isEmpty()).isFalse();
+        String infoLogLevel = "INFO";
```
Comment (geoand) [inline integration-tests/gradle/src/test/java/io/quarkus/gradle/MultiModuleUberJarTest.java:60]: No need to have a variable for this

**Comment by geoand** [comment]
Other than my small comment, above LGTM

File: integration-tests/gradle/src/test/java/io/quarkus/gradle/MultiModuleUberJarTest.java (hunk 4 above)
Comment (Garima829) [reply integration-tests/gradle/src/test/java/io/quarkus/gradle/MultiModuleUberJarTest.java:60]: Okayy

**Comment by geoand** [comment]
Can you squash the commits please?

**Comment by Garima829** [comment]
Yes just a minute

**Review by geoand — APPROVED** [review]
Thanks!

**Pull request merged**